core/src/main/java/org/mule/runtime/core/internal/transaction/TransactionFactoryLocator.java

```diff
@@ -36,12 +37,16 @@
    * @return An {@link Optional} {@link TransactionFactory}
    */
   public Optional<TransactionFactory> lookUpTransactionFactory(TransactionType type) {
+    if (!initialized) {
```

Comment:
I don't understand why you need this flag instead of just having an initialise phase. Assuming there's a good reason for that, you need to reset this on Dispose and use an AtomicBoolean instead, since I'm not sure this class won't be used concurrently